spec/Section 3 -- Type System.md

```diff
@@ -1580,6 +1583,17 @@ Literal Value            | Variables               | Coerced Value
       characters {"__"} (two underscores).
    3. The input field must accept a type where {IsInputType(inputFieldType)}
       returns {true}.
+   4. If the input field has a non-null default value:
```

Comment:
Should this just be:

```suggestion
   4. If the input field has a default value:
```

Since a `null` default value might fail the inner step 2, "must be compatible with {inputFieldType}" if that type might be a Non-Null

Reply:
The `non-null` is here because if the default for the field is `null` (assuming this is compatible with `inputFieldType`), then no further validation is required since no recursion through the default value can occur. Consider A vs B in the below scenario; both are fine, but the validation of A can stop at the `A.a` field immediately, whereas the validation of `B` has to traverse through C and D to get back to finding that the nested reference to `B` via `D.b` defaults to `null`.

```graphql
input A {
  a: A = null
}

input B {
  c: C = {}
}

input C {
  d: D = {}
}

input D {
  b: B = null
}
```

Reply:
I'm less concerned about the recursion (nested step 1) and more about the type correctness (nested step 2)

    4. If the input field has a non-null default value:
      ...
      2. {defaultValue} must be compatible with {inputFieldType} as per the
         coercion rules for that type.

This implies that for the example:

```
input Test {
  a: String! = null
}
```

when validating input field `a`, it would see that the default value is `null` and not progress to determine if it is compatible with `String!` (which it would and should fail)

Reply:
This is no longer relevant with the rewriting of the text I think.